modin/pandas/io.py

```diff
@@ -128,6 +152,158 @@ def read_csv(
 ):  # noqa: PR01, RT01, D200
     """
     Read a comma-separated values (csv) file into DataFrame.
+
+    Parameters Support Status:
```

Comment:
@Garra1980 @alexander3774 what do you think about dynamically add these section? Having this information is great, but it makes the code harder to read.

Reply:
We can go even further and put these tables in special Modin objects (which will store the tables in themselves), and in these docstrings leave information about how they can be viewed.

Reply:
what do you mean - dynamically, can you give some example?

Reply:
this sounds a bit complicated to me... I'd agree on putting the tables in some separate place but on the other hand - I think read_csv's table size is a rare exception

Reply:
placing the parameters tables away from the function it describes will throw us back to the initial state when the tables are unsynced with the actual parameters supported, as it's not near the code and would less likely to be found & updated when the changes are made in the function. And this is actually the problem @alexander3774  is trying to solve with these tables built-in the function's docstrings :smile:

Reply:
:)

Reply:
Yes, `read_csv` docstring is really became big enough, but i think it is an exception rather than regular situation and putting parameters tables away from the implementations will probably leads to the tables outdating.

Reply:
> placing the parameters tables away from the function it describes will throw us back to the initial state when the tables are unsynced with the actual parameters supported, as it's not near the code and would less likely to be found & updated when the changes are made in the function. And this is actually the problem @alexander3774 is trying to solve with these tables built-in the function's docstrings 😄

Then this is not a very obvious reason, I thought it was for the convenience of users (built-in `help` command, for example). It would be nice to add more details to the issue.

Will the docstring be easy to read if we add a couple more columns?

As a reminder, we can leave the following piece of code (reminder), which will be replaced with a table: `%Keep information from docs["read_csv"] up to date%`

Reply:
> Yes, `read_csv` docstring is really became big enough, but i think it is an exception rather than regular situation and putting parameters tables away from the implementations will probably leads to the tables outdating.

Yes, but there is a question of table growth in breadth. To speed up the process (to get into the next release), I don't mind leaving everything as it is, because it's still better than it was.

Reply:
> > placing the parameters tables away from the function it describes will throw us back to the initial state when the tables are unsynced with the actual parameters supported, as it's not near the code and would less likely to be found & updated when the changes are made in the function. And this is actually the problem @alexander3774 is trying to solve with these tables built-in the function's docstrings 😄
> 
> Then this is not a very obvious reason, I thought it was for the convenience of users (built-in `help` command, for example). It would be nice to add more details to the issue.

No, the original idea was to ease tracking and improve visualization of pandas API support - i.e. developer (and reviewer) is responsible for changing information about function's level of API support (which is why those tables are near functions basically) and it will be automatically reflected in the user documentation

Reply:
@anmyachev i agree that these tables can became really big both horiazonatally and vertically, but it's hard to predict now these movements as well as whether docstrings with supported APIs data placeholder will be tracked by developers. I think we can leave this approcah for now, and apply logic with docstrings placeholders if it will be needed. What do you think?

Reply:
ok